docs/sources/shared/troubleshoot-query.md

```diff
@@ -242,3 +242,365 @@ The `count_over_time` function doesn't use unwrapped values - it just counts log
+
+   ```logql
+   # Too broad
+   {job="app"}
+   
+   # More specific
+   {job="app", environment="production", namespace="api"}
+   ```
+
+* **Reduce the query time range** to scan fewer chunks.
+* **Add line filters** to reduce processing:
+
+   ```logql
+   {app="foo"} |= "error"
+   ```
+
+* **Increase the limit** if resources allow:
+
+   ```yaml
+   limits_config:
+     max_chunks_per_query: 5000000  #default is 2000000
+   ```
+
+**Properties:**
+
+- Enforced by: Query Frontend
+- Retryable: No (query must be modified)
+- HTTP status: 400 Bad Request
+- Configurable per tenant: Yes
+
+### Error: Stream matcher limits
+
+**Error message:**
+
+`max streams matchers per query exceeded, matchers-count > limit (1000 > 500)`
+
+**Cause:**
+
+The query contains too many stream matchers. This limit prevents queries with excessive complexity that could impact query performance.
+
+**Default configuration:**
+
+- `max_streams_matchers_per_query`: 1000
+
+**Resolution:**
+
+* **Simplify your query** by using fewer label matchers.
+* **Combine multiple queries** instead of using many OR conditions.
+* **Use regex matchers** to consolidate multiple values:
+
+   ```logql
+   # Instead of many matchers
+   {job="app1"} or {job="app2"} or {job="app3"}
+   
+   # Use regex
+   {job=~"app1|app2|app3"}
+   ```
+
+* **Increase the limit** if needed:
+
+   ```yaml
+   limits_config:
+     max_streams_matchers_per_query: 2000  #default is 1000
+   ```
+
+**Properties:**
+
+- Enforced by: Query Frontend
+- Retryable: No (query must be modified)
+- HTTP status: 400 Bad Request
+- Configurable per tenant: Yes
+
+### Error: Query too large for single querier
+
+**Error message:**
+
+`query too large to execute on a single querier: (query: <size>, limit: <limit>); consider adding more specific stream selectors, reduce the time range of the query, or adjust parallelization settings`
+
+Or for un-shardable queries:
+
+`un-shardable query too large to execute on a single querier: (query: <size>, limit: <limit>); consider adding more specific stream selectors or reduce the time range of the query`
+
+**Cause:**
+
+Even after query splitting and sharding, individual query shards exceed the per-querier byte limit.
+
+**Default configuration:**
+
+- `max_querier_bytes_read`: 150GB (per querier)
+
+**Resolution:**
+
+* **Add more specific stream selectors**.
+* **Reduce the time range** or Break large queries into smaller time ranges.
+* **Simplify the query** if possible - some queries cannot be sharded.
+* **Increase the limit** (requires more querier resources):
+
+   ```yaml
+   limits_config:
+     max_querier_bytes_read: 200GB  # default is 150GB
+   ```
+
+* **Scale querier resources**
+
+**Properties:**
+
+- Enforced by: Query Frontend
+- Retryable: No (query must be modified)
+- HTTP status: 400 Bad Request
+- Configurable per tenant: Yes
+
+### Error: Limit reached while evaluating query
+
+**Error message:**
+
+`limit reached while evaluating the query`
```

Comment:
AFAICT from looking at the code, doen't look like we will ever return this.

it's defined here
https://github.com/grafana/loki/blob/e6df9af776edf43583c10e32d6aa34085ce44d5a/pkg/logqlmodel/error.go#L15

But used only for the series limit
https://github.com/grafana/loki/blob/e6df9af776edf43583c10e32d6aa34085ce44d5a/pkg/logqlmodel/error.go#L90-L103

Reply:
Removed